go/pkg/cas/upload.go

```diff
@@ -894,3 +900,21 @@ func marshalledRequestSize(d *repb.Digest) int64 {
 	}
 	return marshalledFieldSize(reqSize)
 }
+
+// withPerCallTimeout returns a function wrapper that cancels the context if
+// fn does not return within the timeout.
+func withPerCallTimeout(ctx context.Context, timeout time.Duration) (context.Context, context.CancelFunc, func(fn func())) {
```

Comment:
I think this should go either on `cas/client.go` for now, or `client/client.go` where there's already something similar: https://github.com/bazelbuild/remote-apis-sdks/blob/5121020d2a436fe8ced44ab65be31d5f0c4a1276/go/pkg/client/client.go#L694

If you put it on `client/client.go`, it probably should wrap a `func(ctx context.Context)` and use a `context.WithTimeout`, as to be more generic (like the existing `CallWithTimeout`).

Reply:
I've moved it to `cas/client.go`. 

I am still trying to avoid a dependency on `client` package, so that `client` package can use `cas` package. But I agree that this is pretty generic, and we should probably add a new package, at some point, with these generic helper funcs, perhaps an internal package.

`(*Client) CallWithTimeout()` is a bit different though. Yes, both are about timeouts, but this func is specifically a helper for funcs that don't accepted context (like `StreamServer.Send`).